middleware_auth_user.go

```diff
@@ -201,5 +206,25 @@ func (u *userMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next
 		}
 	}
 
+	// We can only use silent reauthentication if:
+	// They present a login cookie (i.e. they're connecting via a browser).
+	// The user manager redirects to a different website (i.e. third-party login
+	// service).
+	// The request method is idempotent (e.g. POST will not work because the
```

Comment:
What happens in the case of POST? Does the request silently fail, or something else?

Reply:
The request will end up at Okta regardless (because evergreen redirects all non-authed users there regardless of the HTTP method), but the payload will be lost. This should be mitigated by the fact that we can use access/refresh tokens, which should reauthorize the user (similarly to how LDAP currently does) without ever needing to log the user in again.

Reply:
SGTM